axlearn/common/flash_attention/tpu_attention.py

```diff
@@ -269,24 +274,11 @@ def _tpu_splash_attention(
     if mask is None:
         mask = splash_attention_mask.FullMask(mask_shape)
     else:
-
-        def wrap_mask(mask: MaskFn) -> MaskFn:
-            """Wrap `mask` so that the return type is a numpy array
-            if the original input was, even if we are inside of jit.
-            """
-
-            def wrapped_mask(*args, **kwargs) -> Union[np.ndarray, Tensor]:
-                if all(
-                    isinstance(x, np.ndarray) for x in jax.tree_util.tree_leaves([args, kwargs])
-                ):
-                    with jax.ensure_compile_time_eval():
-                        result = mask(*args, **kwargs)
-                        return jax.tree.map(np.asarray, result)
-                return mask(*args, **kwargs)
-
-            return wrapped_mask
-
-        mask = ComputableMask(mask_shape, mask_function=wrap_mask(mask))
+        rows = np.arange(source_len)
+        cols = np.arange(target_len)
+        with jax.ensure_compile_time_eval():
+            mask_array = np.asarray(mask(rows[:, None], cols[None, :]))
+        mask = splash_attention_mask.NumpyMask(array=mask_array)
```

Comment:
Comment on the maximum size of this NumpyMask? E.g., it will contain 1B entries for seq_len=32K?

Reply:
Yes indeed, it could consume a large amount of host memory at compile time. Added to the comment. I also added a test to compile with long sequences